Example/ExampleAPI/Endpoint/BookListEndpoint.swift

```diff
@@ -12,10 +12,10 @@ public struct BookListEndpoint: JsonEndpoint {
     
     public typealias Content = [Book]
 
+    
     public init() {}
     
-    public func makeRequest() throws -> URLRequest {
-        return get(URL(string: "books")!)
+    public func makeRequest() -> Result<URLRequest, Error> {
```

Comment:
implicit cast `Error` to `Failure` type

Reply:
The return type should be `Result<URLRequest, Failure>`.

Reply:
The `makeRequest` method of `Endpoint` protocol returns `Result<URLRequest, Failure>` but `makeRequest` method of `BookListEndpoint` returns `Result<URLRequest, Error>`